app/pb/google/api/httpbody.proto

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package google.api;
+
+import "google/protobuf/any.proto";
+
+option cc_enable_arenas = true;
+option go_package = "google.golang.org/genproto/googleapis/api/httpbody;httpbody";
+option java_multiple_files = true;
+option java_outer_classname = "HttpBodyProto";
+option java_package = "com.google.api";
+option objc_class_prefix = "GAPI";
```

Comment:
Please keep our protocol definition language-agnostic.

Would it be possible to draw in these files via pip instead of copying them?

I feel reluctant to add third party files.

Reply:
Indeed, these are copied directly from the google-api definitions [here](https://github.com/googleapis/googleapis). There is a python package that provides these [here](https://pypi.org/project/googleapis-common-protos/) ([github](https://github.com/googleapis/api-common-protos)), but it seems out of date/out of sync with the googleapis repo. Also, I believe we need the full `.proto` definitions in order to get the full descriptor set for envoy, so we would need to get the protos into the `protoc` include path.

For me, it seemed that this sin was justified and much simpler than trying to get the dependencies easily installable across platforms. It's all standalone protobufs anyway, I don't see envoy changing their transcoding definitions any time soon.

We could look for another option, but I don't think it's worth the hassle at the moment, there's a lot of other important things to work on!

(These lines don't restrict language compatibility, they just define some extra options for how the code should be generated for certain languages.)

(Also, small note: I've modified the `include`s here to include from our `pb/` folder.)